tidy3d/web/s3utils.py

```diff
@@ -186,23 +186,26 @@ def _call_back(bytes_in_chunk):
             )
 
 
-def download_file(resource_id: str, remote_filename: str, to_file: str = None):
+def download_file(resource_id: str, remote_filename: str, to_file: str = None, show_progress=True):
```

Comment:
add callback interface to get the progress update information when this api is called by the customer api sdk

Reply:
def download_file(resource_id: str, remote_filename: str, to_file: str = None, show_progress=True, progress_callback=None):
if progress_callback is None, then use our default one defined by show_progress.

Reply:
@lei-flex do we want to provide this option to upload_string and upload_file as well?

Reply:
looks like yes, from your comment below. ok

Reply:
@lei-flex do we want this progress_calback accessible through the main webapi functions (eg upload)? I assume not.

Reply:
I will open it to users if user won't integrate it with their own system. 
so there should be a way to know the upload progress from our interface.:)

Reply:
sorry, I dont understand, so should we add it as a kwarg to webapi upload, download, etc?

or are you saying that users can just call the s3utils functions directly?